src/Features/Core/Portable/ExtractClass/AbstractExtractClassRefactoringProvider.cs

```diff
@@ -55,28 +58,38 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
         private async Task<ExtractClassWithDialogCodeAction?> TryGetMemberActionAsync(CodeRefactoringContext context, IExtractClassOptionsService optionsService)
         {
-            var selectedMemberNode = await GetSelectedNodeAsync(context).ConfigureAwait(false);
-            if (selectedMemberNode is null)
+            var selectedMemberNodes = await GetSelectedNodesAsync(context).ConfigureAwait(false);
+            if (selectedMemberNodes.IsEmpty)
             {
                 return null;
             }
 
             var (document, span, cancellationToken) = context;
             var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var selectedMember = semanticModel.GetDeclaredSymbol(selectedMemberNode, cancellationToken);
-            if (selectedMember is null || selectedMember.ContainingType is null)
+            var memberNodeSymbolPairs = selectedMemberNodes
+                .Select(m => (node: m, symbol: semanticModel.GetDeclaredSymbol(m, cancellationToken)))
+                // Use same logic as pull members up for determining if a selected member
+                // is valid to be moved into a base
+                .Where(pair => pair.symbol != null && MemberAndDestinationValidator.IsMemberValid(pair.symbol))
+                .AsImmutable();
+
+            var selectedMembers = memberNodeSymbolPairs.SelectAsArray(pair => pair.symbol!);
```

Comment:
We pair each node to its symbol because if we select multiple member nodes but only a subset of them are valid, we want to make sure we take all the valid ones (as opposed to failing to refactor, or keeping the incorrect nodes). This behavior and some other behavior such as getting the containing type is mostly similar across the other two other member refactoring services as well. It could possibly be refactored into common method(s) but I'm not sure where it would go.